pkg/loadbalancer/experimental/bpf_reconciler.go

```diff
@@ -184,7 +184,7 @@ type bpfOpsParams struct {
 }
 
 func newBPFOps(p bpfOpsParams) *BPFOps {
-	if !p.Cfg.EnableExperimentalLB {
+	if !(p.Cfg.EnableExperimentalLB || p.ExtCfg.LoadBalancerAlgorithmAnnotation) {
```

Comment:
This won't work. The `EnableExperimentalLB` is covering the whole new control-plane, so we can't have a condition like this as then we'd be running the old and new at the same time.

Note that for v1.18 the old one will be removed and this `EnableExperimentalLB` flag will go away.